asserts/asserts.go

```diff
@@ -218,11 +220,173 @@ func Decode(serializedAssertion []byte) (Assertion, error) {
 		return nil, fmt.Errorf("parsing assertion headers: %v", err)
 	}
 
-	if len(signature) == 0 {
-		return nil, fmt.Errorf("empty assertion signature")
+	return Assemble(headers, body, content, signature)
+}
+
+// Maximum assertion component sizes.
+const (
+	MaxBodySize      = 2 * 1024 * 1024
+	MaxHeadSize      = 128 * 1024
+	MaxSignatureSize = 128 * 1024
+)
+
+// Decoder parses a stream of assertions bundled by ensuring double newlines at the end of each assertion.
+type Decoder struct {
+	rd             io.Reader
+	initialBufSize int
+	b              *bufio.Reader
+	err            error
+	maxHeadSize    int
+	maxBodySize    int
+	maxSigSize     int
+}
+
+func newDecoder(r io.Reader, bufSize, maxHeadSize, maxBodySize, maxSigSize int) *Decoder {
```

Comment:
This doesn't look like an improvement over the logic inside the function. Any time we use it we'll have to check what order the fields are in. I suggest just sticking to the type itself.